src/main/java/de/zalando/aruha/nakadi/service/ClosedConnectionsCrutch.java

```diff
@@ -0,0 +1,294 @@
+package de.zalando.aruha.nakadi.service;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.net.InetAddresses;
+import de.zalando.aruha.nakadi.util.FeatureToggleService;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.StringTokenizer;
+import java.util.TreeSet;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BooleanSupplier;
+import java.util.stream.Collectors;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.commons.lang3.ArrayUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ClosedConnectionsCrutch {
+    /**
+     * List of allowed connection states
+     * http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/include/net/tcp_states.h
+     */
+    enum ConnectionState {
+        TCP_ESTABLISHED(1),
+        TCP_SYN_SENT(2),
+        TCP_SYN_RECV(3),
+        TCP_FIN_WAIT1(4),
+        TCP_FIN_WAIT2(5),
+        TCP_TIME_WAIT(6),
+        TCP_CLOSE(7),
+        TCP_CLOSE_WAIT(8),
+        TCP_LAST_ACK(9),
+        TCP_LISTEN(10),
+        TCP_CLOSING(11),	/* Now a valid state */
+        TCP_NEW_SYN_RECV(12),;
+        private final int stateCode;
+
+        ConnectionState(final int stateCode) {
+            this.stateCode = stateCode;
+        }
+
+        public static ConnectionState fromCode(final String value) {
+            final int intCode = Integer.parseInt(value, 16);
+            for (final ConnectionState val : ConnectionState.values()) {
+                if (val.stateCode == intCode) {
+                    return val;
+                }
+            }
+            throw new IllegalArgumentException("Failed to find connection state from " + value);
+        }
+    }
+
+    static class ConnectionInfo {
+        private final InetAddress address;
+        private final int port;
```

Comment:
You identify connection by IP and port. What if there are several connections from a machine to Nakadi? I think it may be quite a common case that the client-app opens several connections. (or that several client-apps are running at the same machine)